internal/clients/peering.go

```diff
@@ -48,25 +51,34 @@ func peeringRefreshState(ctx context.Context, client *Client, peeringID string,
 	}
 }
 
-// WaitForPeeringToBePendingAcceptance will poll the GET peering endpoint until
-// the state is PENDING_ACCEPTANCE, ctx is canceled, or an error occurs.
-func WaitForPeeringToBePendingAcceptance(ctx context.Context, client *Client, peeringID string, hvnID string, loc *sharedmodels.HashicorpCloudLocationLocation, timeout time.Duration) (*networkmodels.HashicorpCloudNetwork20200907Peering, error) {
-	stateChangeConf := resource.StateChangeConf{
-		Pending: []string{
-			PeeringStateCreating,
-		},
-		Target: []string{
-			PeeringStatePendingAcceptance,
-		},
-		Refresh:      peeringRefreshState(ctx, client, peeringID, hvnID, loc),
-		Timeout:      timeout,
-		PollInterval: 5 * time.Second,
-	}
+type WaitFor = func(ctx context.Context, client *Client, peeringID string, hvnID string, loc *sharedmodels.HashicorpCloudLocationLocation, timeout time.Duration) (*networkmodels.HashicorpCloudNetwork20200907Peering, error)
 
-	result, err := stateChangeConf.WaitForStateContext(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("Error waiting for network peering (%s) to become 'PENDING_ACCEPTANCE': %s", peeringID, err)
-	}
+func waitForPeeringToBe(state string) WaitFor {
```

Comment:
Ha I'm not sure I knew go did closures. 💡 Why'd you decide to use one here?

Reply:
I used it here because I needed the same function to check for two different peering states and the amount of params in that function was already on the longer side to add the state there. This seemed to be the most pragmatic and clean approach